### PR TITLE
Staking / Ursula guide update

### PR DESCRIPTION
**Pull request by michwill**
Fixing documentation in accordance with the issues found

### CODE REVIEW

File: docs/source/guides/staking_guide.rst [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
 NuCypher Staking Guide
 =======================
 
-All staking-related operations are performed through the ``nucypher stake`` command:
+The account which is actively doing staking operations is necessary to be a hot
+wallet. But it doesn't have to be the same account as the one which receives and
+sends NU tokens. The account which holds NU tokens is called StakeHolder, or
+Staker, and the account which participates in the network as Ursula node is
+called Worker.
+StakeHolder and Worker can have the same Ethereum account, but it's better to
```
Comment (cygnusv) [inline docs/source/guides/staking_guide.rst]: I'd rephrase this to make it in a less positive light, saying first what is recommended (staker != worker), and only later saying that staker==worker is allowed, but not recommended.

File: docs/source/guides/staking_guide.rst [hunk 2]
```diff
@@ -1,8 +1,19 @@
+.. _staking-guide:
+
 =======================
 NuCypher Staking Guide
 =======================
 
-All staking-related operations are performed through the ``nucypher stake`` command:
+The account which is actively doing staking operations is necessary to be a hot
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]: What are we defining as "staking operations" here? Staking operations as I understand them (e.g. stake, split stake, etc.) do not require a hot wallet. If you're referring to Worker operations (e.g. re-encrypt) I'd call them something else.

Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]:
```suggestion
The account which is actively doing staking operations needs to be a hot
```

File: docs/source/guides/staking_guide.rst [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+wallet. But it doesn't have to be the same account as the one which receives and
+sends NU tokens. The account which holds NU tokens is called StakeHolder, or
+Staker, and the account which participates in the network as Ursula node is
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]:
```suggestion
Staker, and the account which participates in the network as an Ursula node is
```

File: docs/source/guides/staking_guide.rst [hunk 4]
```diff
@@ -53,31 +68,71 @@ Staking Overview
...
-2) Install and run Geth, Parity or another ethereum node.
+1) Install nucypher on StakeHolder node https://docs.nucypher.com/en/latest/guides/installation_guide.html
+2) Install and run Geth, Parity or another ethereum node (can use with software or hardware Ethereum wallet).
 3) Request testnet tokens from faucet
 4) Stake tokens (See Below)
-5) Initialize a Worker node and bond it to your Staker (`set-worker`) https://docs.nucypher.com/en/latest/guides/ursula_configuration_guide.html
-6) Run the Worker, and keep it online!
+5) Install another Ethereum node at the Worker instance
+6) Initialize a Worker node and bond it to your Staker (``set-worker``) :ref:`ursula-config-guide`
+7) Configure and run the Worker, and keep it online (See Below)!
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]: make See Below a relative url

File: docs/source/guides/staking_guide.rst [hunk 5]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
     | ~ | Staker | Worker | # | Value    | Duration     | Enactment
     |   | ------ | ------ | - | -------- | ------------ | -----------------------------------------
-    | 0 | 0xbb01 | 0xbb02 | 0 | 15000 NU | 41 periods . | Aug 04 12:15:16 CEST - Sep 13 12:15:16 CEST
-    | 1 | 0xbb01 | 0xbb02 | 1 | 15000 NU | 30 periods . | Aug 20 12:15:16 CEST - Sep 18 12:15:16 CEST
-    | 2 | 0xbb03 |    -   | 0 | 30000 NU | 30 periods . | Aug 09 12:15:16 CEST - Sep 9 12:15:16 CEST
+    | 0 | 0xbb01 | 0xdead | 0 | 15000 NU | 41 periods . | Aug 04 12:15:16 CEST - Sep 13 12:15:16 CEST
+    | 1 | 0xbb02 | 0xbeef | 1 | 15000 NU | 30 periods . | Aug 20 12:15:16 CEST - Sep 18 12:15:16 CEST
+    | 2 | 0xbb03 | 0x0000 | 0 | 30000 NU | 30 periods . | Aug 09 12:15:16 CEST - Sep 9 12:15:16 CEST
+
+If the Worker in the list is shown as ``0x0000``, it means that you didn't yet
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]:
```suggestion
If the Worker in the list is shown as ``0x0000``, it means that you haven't yet
```

File: docs/source/guides/staking_guide.rst [hunk 6]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
+attach any node to your Staker, so you still have to do it!
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]:
```suggestion
attached a Worker node to your Staker, so you still have to do it!
```

File: docs/source/guides/staking_guide.rst [hunk 7]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
 
 
-*Bond an Ursula to a Staker*
+Bond an Ursula to a Staker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After you created an Ethereum node for your worker (see below about the worker),
+you can set the worker. The stake which doesn't yet have a worker will be
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]:
```suggestion
you can set the worker. Stake which don't yet have a worker will be
```

File: docs/source/guides/staking_guide.rst [hunk 8]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
 
     Select Stake: 2
-    Enter Worker Address: 0xbb04c4fE50f91eF73c5dD6eD89f38D55A6b1EdCA
+    Enter Worker Address: 0xbeefc4fE50f91eF73c5dD6eD89f38D55A6b1EdCA
     Worker 0xbb04c4fE50f91eF73c5dD6eD89f38D55A6b1EdCA successfully bonded to staker 0xbb03...
 
     OK!
 
+Please note that the address should be in the format where checksum is encoded
+in the address. However, geth shows addresses in the lower case. You can convert
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]: Is it easier for us to just accept lower case addresses rather than asking people to manually convert by copy/paste?

File: docs/source/guides/staking_guide.rst [hunk 9]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
+After this step, you're finished with the Staker, and you can proceed to :ref:`ursula-config-guide`.
+
+
+Collect rewards earned by the staker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Choose your staking address to withdraw compensation from
+(``nucypher stake accounts`` will show). Then, use
+``nucypher stake collect-reward`` with an option ``--staking-reward`` to collect
+inflation rewards in NU or ``--policy-reward`` to collect Ethers earned as
```
Comment (mswilkison) [inline docs/source/guides/staking_guide.rst]: If you leave out the flag, does it collect both staking and policy rewards?

File: docs/source/guides/ursula_configuration_guide.rst [hunk 10]
```diff
@@ -1,9 +1,55 @@
+.. _ursula-config-guide:
+
+==========================
 Ursula Configuration Guide
 ==========================
 
+1. Install NuCypher and geth
+------------------------------
+
+If you want to run a NuCypher node which participates in decentralized network,
```
Comment (mswilkison) [inline docs/source/guides/ursula_configuration_guide.rst]:
```suggestion
If you want to run a NuCypher node that participates in the decentralized network,
```

File: docs/source/guides/staking_guide.rst (hunk 2 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: True, that's only stakeholder's commands. I thought to put worker commands here, but I realized that there is already a section for that. Will fix

File: docs/source/guides/staking_guide.rst (hunk 8 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: I think so! We can do the same as we did for Felix

File: docs/source/guides/staking_guide.rst (hunk 9 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: Eh, right now it's a _required_ flag. Also not sure if it collects both if I put both - need to try.
I think, it's a sensible default to collect both without the flag, right?

File: docs/source/guides/staking_guide.rst (hunk 1 above)
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]:
```suggestion
While it is possible for the StakeHolder and Worker to have the same Ethereum account, we strongly recommend keeping them separate:
```

File: docs/source/guides/staking_guide.rst [hunk 11]
```diff
@@ -1,8 +1,19 @@
...
 =======================
 
-All staking-related operations are performed through the ``nucypher stake`` command:
+The account which is actively doing staking operations is necessary to be a hot
+wallet. But it doesn't have to be the same account as the one which receives and
+sends NU tokens. The account which holds NU tokens is called StakeHolder, or
+Staker, and the account which participates in the network as Ursula node is
+called Worker.
+StakeHolder and Worker can have the same Ethereum account, but it's better to
+have them separate: StakeHolder controlled by a hardware wallet, and Worker
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]: Associated with previous suggestion
```suggestion
 StakeHolder controlled by a hardware wallet, and Worker
```

File: docs/source/guides/staking_guide.rst [hunk 12]
```diff
@@ -53,31 +68,71 @@ Staking Overview
 
 Most stakers on the Goerli testnet will complete the following steps:
 
-1) Install nucypher https://docs.nucypher.com/en/latest/guides/installation_guide.html
-2) Install and run Geth, Parity or another ethereum node.
+1) Install nucypher on StakeHolder node https://docs.nucypher.com/en/latest/guides/installation_guide.html
+2) Install and run Geth, Parity or another ethereum node (can use with software or hardware Ethereum wallet).
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]:
```suggestion
2) Install and run Geth, Parity or another ethereum node (can be used with software or hardware Ethereum wallet).
```

File: docs/source/guides/staking_guide.rst [hunk 13]
```diff
@@ -53,31 +68,71 @@ Staking Overview
...
+7) Configure and run the Worker, and keep it online (See Below)!
 
 Interactive Method
 ------------------
 
-*Initialize a new stakeholder*
+Run an Ethereum node for stakeholder
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Assuming you have ``geth`` installed, let's run a node on Görli testnet.
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]: We should be consistent with our spelling - we have used both `Goerli` and `Görli`.

File: docs/source/guides/staking_guide.rst [hunk 14]
```diff
@@ -134,37 +192,86 @@ Interactive Method
...
+
+If the Worker in the list is shown as ``0x0000``, it means that you didn't yet
+attach any node to your Staker, so you still have to do it!
 
 
-*Bond an Ursula to a Staker*
+Bond an Ursula to a Staker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After you created an Ethereum node for your worker (see below about the worker),
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]:
```suggestion
After you create an Ethereum node for your worker (see below about the worker),
```

File: docs/source/guides/staking_guide.rst (hunk 7 above)
Comment (derekpierre) [reply docs/source/guides/staking_guide.rst]: ^ _**Stakes**_ which don't yet...

File: docs/source/guides/staking_guide.rst [hunk 15]
```diff
@@ -218,17 +325,24 @@ Inline Method
...
 .. code:: bash
 
-    (nucypher)$ nucypher stake divide --index 0 --value 15000 --duration 30
+    (nucypher)$ nucypher stake divide --index 0 --value 15000 --duration 30 --hw-wallet
     ...
+
+Worker configuration
+------------------------
+
+Please go to :ref:`ursula-config-guide`.
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]:
```suggestion
See :ref:`ursula-config-guide`.
```

File: docs/source/guides/ursula_configuration_guide.rst [hunk 16]
```diff
@@ -1,9 +1,55 @@
...
+Install ``nucypher`` either by doing ``sudo pip3 install nucypher`` if you have
+a dedicated instance or container, or with a ``virtualenv``:
+
+.. code:: bash
+
+    $ virtualenv -p python3 nucypher
+    $ source nucypher/bin/activate
+    (nu)$ pip3 install nucypher
+
 Before continuing, Verify your ``nucypher`` installation and entry points are functional:
```
Comment (derekpierre) [inline docs/source/guides/ursula_configuration_guide.rst]:
```suggestion
Before continuing, verify that your ``nucypher`` installation and entry points are functional:
```

File: docs/source/guides/ursula_configuration_guide.rst [hunk 17]
```diff
@@ -1,9 +1,55 @@
...
 
-Activate your virtual environment and run the ``nucypher --help`` command
+Activate your virtual environment (if you already didn't) and run the ``nucypher --help`` command
```
Comment (derekpierre) [inline docs/source/guides/ursula_configuration_guide.rst]:
```suggestion
Activate your virtual environment (if you haven't already) and run the ``nucypher --help`` command
```

File: docs/source/guides/staking_guide.rst (hunk 9 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: Ok, looks like if you specify both, it actually works ok

File: docs/source/guides/staking_guide.rst (hunk 13 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: Yeah, ok, `oe` is kind of a way to spell when you cannot type `ö`. I assume, can be interchangeable?

File: docs/source/guides/staking_guide.rst (hunk 9 above)
Comment (michwill) [reply docs/source/guides/staking_guide.rst]: :heavy_check_mark:

File: docs/source/guides/staking_guide.rst [hunk 18]
```diff
@@ -134,37 +190,93 @@ Interactive Method
...
     |   | ------ | ------ | - | -------- | ------------ | -----------------------------------------
-    | 0 | 0xbb01 | 0xbb02 | 0 | 15000 NU | 41 periods . | Aug 04 12:15:16 CEST - Sep 13 12:15:16 CEST
-    | 1 | 0xbb01 | 0xbb02 | 1 | 15000 NU | 30 periods . | Aug 20 12:15:16 CEST - Sep 18 12:15:16 CEST
-    | 2 | 0xbb03 |    -   | 0 | 30000 NU | 30 periods . | Aug 09 12:15:16 CEST - Sep 9 12:15:16 CEST
+    | 0 | 0xbb01 | 0xdead | 0 | 15000 NU | 41 periods . | Aug 04 12:15:16 CEST - Sep 13 12:15:16 CEST
+    | 1 | 0xbb02 | 0xbeef | 1 | 15000 NU | 30 periods . | Aug 20 12:15:16 CEST - Sep 18 12:15:16 CEST
+    | 2 | 0xbb03 | 0x0000 | 0 | 30000 NU | 30 periods . | Aug 09 12:15:16 CEST - Sep 9 12:15:16 CEST
+
+If the Worker in the list is shown as ``0x0000``, it means that you haven't yet
+attach attached a Worker node to your Staker, so you still have to do it!
```
Comment (derekpierre) [inline docs/source/guides/staking_guide.rst]:
```suggestion
attached a Worker node to your Staker, so you still have to do it!
```

**Comment by michwill** [comment]
Any final suggestions?
